src/app/psbt.ts

```diff
@@ -331,6 +338,167 @@ const Psbt = function (config) {
+            const spendableAmount = nonOrdinals.reduce((acc, utxo) => acc + utxo.value, 0);
+
+            const metadata: Metadata = {
+                inputs: inputs.map((utxo, index) => ({
+                    index,
+                    type: utxo.inscriptionId ? 'Ordinal' : 'Cardinal',
+                })),
+                outputs: [],
+            };
+
+            // Add ordinal outputs
+            ordinals.forEach(utxo => {
+                psbt.addOutput({
+                    address: destinationBtcAddress,
+                    value: utxo.value,
+                });
+                metadata.outputs.push({ type: 'Ordinal' });
+            });
+
+            let changeAmount = 0;
+
+            // Calculate and add change output if needed
+            if (isCardinalAdded) {
+                changeAmount = cardinalAmount - calculatedFee;
+                psbt.addOutput({
+                    address,
+                    value: changeAmount,
+                });
+                metadata.outputs.push({ type: 'Change' });
+            }
+
+            // Calculate and add destination output
+            const destinationOutputValue = spendableAmount - changeAmount - calculatedFee;
+            if (destinationOutputValue > 0) {
+                psbt.addOutput({
+                    address: destinationBtcAddress,
+                    value: destinationOutputValue,
+                });
+                metadata.outputs.push({ type: 'Cardinal' });
+            }
+
+            const witnessScripts = [];
+            const witnessValues = [];
+
+            psbt.data.inputs.forEach((input, i) => {
+                if (!input.finalScriptWitness && !input.witnessUtxo) {
+                    // @ts-ignore
+                    const tx = bitcoin.Transaction.fromBuffer(psbt.data.inputs[i].nonWitnessUtxo);
+                    const output = tx.outs[psbt.txInputs[i].index];
+                    psbt.updateInput(i, {
+                        witnessUtxo: output,
+                    });
+                    // @ts-ignore
+                    witnessScripts.push(output.script);
+                    // @ts-ignore
+                    witnessValues.push(output.value);
+                } else {
+                    // @ts-ignore
+                    witnessScripts.push(psbt.data.inputs[i].witnessUtxo.script);
+                    // @ts-ignore
+                    witnessValues.push(psbt.data.inputs[i].witnessUtxo.value);
+                }
+            });
+
+            const psbtOutputs = psbt.data.inputs.map((_, index) => {
+                // @ts-ignore
+                const sigHash = psbt.__CACHE.__TX.hashForWitnessV1(
+                    index,
+                    witnessScripts,
+                    witnessValues,
+                    bitcoin.Transaction.SIGHASH_DEFAULT
+                );
+
+                return {
+                    index,
+                    sigHash: sigHash.toString('hex')
+                };
+            });
+
```

Comment:
The calcualted fee will be very close to the final fee so we can skip signing before the confirm step.

Reply:
Or maybe just update the final fee shown in the model.

So we show estimated fee first, user hits confirm, then signs stuff, then views the final tx before hitting "send"?